sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -679,20 +679,33 @@ abstract class BaseExprMeta[INPUT <: Expression](
   override def canExprTreeBeReplaced: Boolean =
     canThisBeReplaced && super.canExprTreeBeReplaced
 
+  def dataType: DataType = expr.dataType
+
   final override def tagSelfForGpu(): Unit = {
     try {
       if (!areAllSupportedTypes(expr.dataType)) {
         willNotWorkOnGpu(s"expression ${expr.getClass.getSimpleName} $expr " +
           s"produces an unsupported type ${expr.dataType}")
       }
-    }
-    catch {
+    } catch {
       case _ : java.lang.UnsupportedOperationException =>
         if (!ignoreUnsetDataTypes) {
           willNotWorkOnGpu(s"expression ${expr.getClass.getSimpleName} $expr " +
             s" does not have a corresponding dataType.")
         }
     }
+    val inputDataTypes = childExprs.map { expr =>
+      try {
+        expr.dataType
+      } catch {
+        case _ : java.lang.UnsupportedOperationException => null
```

Comment:
This is something in Spark code that we cannot control.  Expression has an API to get the `dataType` and for some reason  one API decided to throw instead of returning something.  We could try and detect it with reflection and go on, but I think this is cleaner.